src/main/java/it/pagopa/transactions/commands/handlers/TransactionUpdateAuthorizationHandler.java

```diff
@@ -0,0 +1,125 @@
+package it.pagopa.transactions.commands.handlers;
+
+import it.pagopa.generated.ecommerce.nodo.v1.dto.AdditionalPaymentInformationsDto;
+import it.pagopa.generated.ecommerce.nodo.v1.dto.ClosePaymentRequestDto;
+import it.pagopa.generated.transactions.server.model.AuthorizationResultDto;
+import it.pagopa.generated.transactions.server.model.TransactionInfoDto;
+import it.pagopa.generated.transactions.server.model.TransactionStatusDto;
+import it.pagopa.generated.transactions.server.model.UpdateAuthorizationRequestDto;
+import it.pagopa.transactions.client.NodeForPspClient;
+import it.pagopa.transactions.commands.TransactionUpdateAuthorizationCommand;
+import it.pagopa.transactions.documents.TransactionAuthorizationRequestData;
+import it.pagopa.transactions.documents.TransactionAuthorizationRequestedEvent;
+import it.pagopa.transactions.documents.TransactionAuthorizationStatusUpdateData;
+import it.pagopa.transactions.documents.TransactionAuthorizationStatusUpdatedEvent;
+import it.pagopa.transactions.domain.Transaction;
+import it.pagopa.transactions.exceptions.AlreadyProcessedException;
+import it.pagopa.transactions.repositories.TransactionsEventStoreRepository;
+import it.pagopa.transactions.utils.TransactionEventCode;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Component
+@Slf4j
+public class TransactionUpdateAuthorizationHandler implements CommandHandler<TransactionUpdateAuthorizationCommand, Mono<TransactionInfoDto>> {
+
+    @Autowired
+    NodeForPspClient nodeForPspClient;
+
+    @Autowired
+    private TransactionsEventStoreRepository<TransactionAuthorizationStatusUpdateData> transactionEventStoreRepository;
+
+    @Autowired
+    private TransactionsEventStoreRepository<TransactionAuthorizationRequestedEvent> authorizationRequestedEventStoreRepository;
+
+    @Override
+    public Mono<TransactionInfoDto> handle(TransactionUpdateAuthorizationCommand command) {
+        Transaction transaction = command.getData().transaction();
+
+        if (transaction.getStatus() != TransactionStatusDto.AUTHORIZATION_REQUESTED) {
+            log.error("Error: requesting authorization update for transaction in state {}", transaction.getStatus());
+            return Mono.error(new AlreadyProcessedException(transaction.getRptId()));
+        } else {
+            UpdateAuthorizationRequestDto updateAuthorizationRequest = command.getData().updateAuthorizationRequest();
+
+            return authorizationRequestedEventStoreRepository.findByPaymentTokenAndEventCode(
```

Comment:
ok check on the `eventstore` and not on the `view` 🥇  but in a separate story / PR all these operations must be performed by the aggregate